languages/typescript/src/main/java/de/jplag/typescript/grammar/TypeScriptLexerBase.java

```diff
@@ -0,0 +1,142 @@
+package de.jplag.typescript.grammar;
+
+import java.util.ArrayDeque;
+import java.util.Deque;
+
+import org.antlr.v4.runtime.CharStream;
+import org.antlr.v4.runtime.Lexer;
+import org.antlr.v4.runtime.Token;
+
+/**
+ * Copied from https://github.com/antlr/grammars-v4/tree/master/javascript/typescript/Java
+ */
+abstract class TypeScriptLexerBase extends Lexer {
```

Comment:
```suggestion
@SuppressWarnings({"java:S100", "java:S5993", "java:S1155", "java:S6208"})
abstract class TypeScriptLexerBase extends Lexer {
```

I'd suppress warnings of sonarlint for the copied classes. Not nice, but I don't think we want to let these external dependencies count as issue . (but not sure)

Reply:
We decided on modifying the files from the ANTRL Repo to fit the JPlag code style